src/main/java/org/socialhistoryservices/delivery/api/EADRecordExtractor.java

```diff
@@ -152,6 +155,31 @@ public ExternalRecordInfo getRecordMetadata(Node node, String item) throws NoSuc
         return retMap;
     }
 
+    @Override
+    public String getUnitIdsFromContainer(Node node, String pid, String container, boolean includeCurrentContainer) throws NoSuchPidException {
+        ArrayList<String> listOfContainers = new ArrayList<String>();
+        String[] parentPidAndItem = Utils.getParentPidAndItem(pid);
+
+        try {
+            NodeList itemNodes = (NodeList) xpath.evaluate(".//ead:dsc//ead:unitid", node, XPathConstants.NODESET);
+
+            for (int i = 0; i < itemNodes.getLength(); i++) {
+                Node child = itemNodes.item(i);
+
+                if (child.getTextContent().toLowerCase().startsWith(container.toLowerCase() + " ")) { // don't remove space " "
```

Comment:
I don't understand why the unit id has to start with the container? And why is the space " " required? Where is the query for the container?